<h1 style = "text-align: center;"> Phishing prediction </h1>



In [147]:
# importing all the dependencies

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

from sklearn.linear_model import LogisticRegression, LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score


In [148]:
data = pd.read_csv('Phishing_data.csv') # get the data stored as a csv file
# create a data frame in pandas.
df = pd.DataFrame(data)
#print(df)
# converting it to an np array to allow splicing the data frame so that we can extract the features and labels. 
df = np.array(df) 

In [149]:
# features are the first 30 columns. Extracting the first 30 columns in a variable callled features
features = df[:, :len(df[1]) - 1]
#print(features[:10])
# labels are the last column. i.e, the prediction values.
labels = df[:, len(df[1]) - 1:len(df[1])]
labels = labels.reshape(len(labels), )
#print(labels[:10])

#splitting the data into training data and testing data using sklearn
x_train, x_test, y_train, y_test = train_test_split(features, labels, train_size = 0.80, test_size = 0.20, random_state = 0) # splitting the data 

In [150]:
# creating a logistic regression classifier.
logistic = LogisticRegression(verbose = True)
# training the classifier on the train data.
logistic.fit(x_train, y_train)

[LibLinear]

LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
          intercept_scaling=1, max_iter=100, multi_class='ovr', n_jobs=1,
          penalty='l2', random_state=None, solver='liblinear', tol=0.0001,
          verbose=True, warm_start=False)

In [151]:
# making a prediction with the test data.
logistic_pred = logistic.predict(x_test)

In [156]:
# testing the accuracy of the model with sklearn.
accuracy = accuracy_score(logistic_pred, y_test)
print(accuracy*100, "%")

94.10569105691057 %


<h1> Results for Logistic Regression </h1>
<p style="padding-bottom: 10px;">
<ol>
<li> It can be seen that the logistic regression model gives an accuracy of 94% </li>
      <br>
<li> I have used 80% of the data for training and 20% of the data for testing the model. </li>
       <br>

In [158]:
# using linear regression to compare
linear = LinearRegression()
# training the linear model
linear.fit(x_train, y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=1, normalize=False)

In [159]:
# making predictions
linear_pred = linear.predict(x_test)
# rounding the values to represent them as integers. (default = float)
linear_pred = np.round(linear_pred)
# making predictions with the linear model.
linear_pred = linear_pred.astype(int)
#print(linear_pred)

In [161]:
# calculating the accuracy of the linear model
linear_accuracy = accuracy_score(linear_pred, y_test)
print(linear_accuracy*100, "%")

92.47967479674797 %


<h1> Results for Linear Regression </h1>
<p style="padding-bottom: 10px;">
<ol>
<li> It can be seen that the linear regression model gives an accuracy of 92% </li>
      <br>
<li> I have used 80% of the data for training and 20% of the data for testing the model. </li>
       <br>
<li> The Logistic Regression Model performs better than the Linear Model. </li>
       <br>